src/openrct2/interface/window.c

```diff
@@ -856,6 +856,22 @@ void window_close_all_except_class(rct_windowclass cls)
     }
 }
 
+void window_close_all_except_flags(uint16 flags)
+{
+    for (rct_window * w = g_window_list; w < RCT2_NEW_WINDOW;)
+    {
+        if (!(w->flags & flags))
```

Comment:
If you pass more than one flag as argument to this function, having any of them set will trigger this condition. It looks like this is meant behaviour, but it's not clear given the interface, so please add a comment explaining this.